xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -117,6 +117,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
```

Comment:
On 2016/03/25 19:08, Rainer Hochecker <notifications@github.com> wrote:
> see the snippet in my post above: i.e.
> CVideoPlayer::GetStreamDetails or
> CVideoPlayer::GetAudioStreamInfo
> (there is more)

OK, so these methods are called from outside the player thread.  I
thought what you meant was users of CSelectionStreams from classes
other than VideoPlayer.  I see.

I will check these methods and see what I can do to implement lockless
read-only access.  (Out of this PR's scope.)